File: server/api/miners/export.ts
import { getMiners } from "../service/miner";
import { toCsv, toExcel, toJson } from "../lib/export";
export default async function handler(req, res) {
  const miners = await getMiners();
  const { format = "csv" } = req.query;
  if (format === "csv") return res.send(toCsv(miners));
  if (format === "xlsx") return res.sendFile(await toExcel(miners));
  if (format === "json") return res.json(toJson(miners));
}

File: server/api/export.py
import pandas as pd
from fastapi.responses import StreamingResponse

def export_report(format: str):
    df = pd.read_sql("SELECT * FROM miners", db.engine)
    if format == "csv":
        return StreamingResponse(df.to_csv(index=False), media_type="text/csv")
    if format == "json":
        return df.to_json(orient="records")
    if format == "xlsx":
        out = io.BytesIO()
        df.to_excel(out, index=False)
        out.seek(0)
        return StreamingResponse(out, media_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")